widget/embedded/src/pages/HistoryPage.tsx

```diff
@@ -1,47 +1,100 @@
+
 import { i18n } from '@lingui/core';
 import { useManager } from '@rango-dev/queue-manager-react';
-import { Alert, SwapsGroup } from '@rango-dev/ui';
-import React from 'react';
+import {
+  Alert,
+  SearchIcon,
+  styled,
+  SwapsGroup,
+  TextField,
+} from '@rango-dev/ui';
+import React, { useState } from 'react';
 import { useNavigate } from 'react-router-dom';
 
 import { Layout } from '../components/Layout';
 import { navigationRoutes } from '../constants/navigationRoutes';
 import { useNavigateBack } from '../hooks/useNavigateBack';
 import { useUiStore } from '../store/ui';
 import { groupSwapsByDate } from '../utils/date';
+import { containsText } from '../utils/numbers';
 import { getPendingSwaps } from '../utils/queue';
 
+const Container = styled('div', {
+  display: 'flex',
+  flexDirection: 'column',
+  alignItems: 'flex-start',
+  gap: 15,
+});
+
+const List = styled('div', {
+  overflowY: 'auto',
+  width: '100%',
+  display: 'flex',
+  flexDirection: 'column',
+  gap: 15,
+  height: 440,
+  paddingRight: '$8',
+});
+
 export function HistoryPage() {
   const setSelectedSwap = useUiStore.use.setSelectedSwap();
   const navigate = useNavigate();
   const { navigateBackFrom } = useNavigateBack();
   const { manager, state } = useManager();
   const list = getPendingSwaps(manager).map(({ swap }) => swap);
-
+  const [searchedFor, setSearchedFor] = useState<string>('');
+  const [filteredList, setFilteredList] = useState<PendingSwap[]>([]);
   const loading = !state.loadedFromPersistor;
 
+  const filterTransactions = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+    setSearchedFor(value);
+    const filtered = list.filter(
+      (swap) =>
+        containsText(swap.inputAmount, value) ||
+        containsText(swap.status || '', searchedFor)
+    );
+    setFilteredList(filtered);
+  };
+
   return (
     <Layout
-      hasFooter
       header={{
         onBack: navigateBackFrom.bind(null, navigationRoutes.swaps),
         title: i18n.t('History'),
       }}>
       {!loading && (
-        <>
-          {list.length ? (
-            <SwapsGroup
-              list={list}
-              onSwapClick={(requestId) => {
-                setSelectedSwap(requestId);
-                navigate(`${requestId}`, { replace: true });
-              }}
-              groupBy={groupSwapsByDate}
-            />
-          ) : (
-            <Alert type="info" title="Swap" />
-          )}
-        </>
+        <Container>
+          <TextField
+            prefix={<SearchIcon />}
+            fullWidth
+            color="light"
+            variant="contained"
+            placeholder="Search Transaction"
+            style={{
+              padding: 10,
+              borderRadius: 25,
+              alignItems: 'center',
+            }}
+            onChange={filterTransactions}
+            value={searchedFor}
+          />
+          <List>
```

Comment:
It is similar to the list component name in ui,
I think it can have a more unique name

Reply:
I suggest `SwapList`

Reply:
fixed.